src/main/java/TaskList.java

```diff
@@ -0,0 +1,46 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.io.File;
+import java.util.Scanner;
+
+public class TaskList {
+
+  private ArrayList<Task> taskList = new ArrayList<>();
+  private Storage s;
+
+  public TaskList(Storage s) throws FileNotFoundException, DukeException, IOException {
+    try {
+      this.s = s;
+      Scanner sc = new Scanner(s.load());
+      while (sc.hasNextLine()) {
+        String command = sc.next();
+        String description = sc.nextLine().stripLeading();
+
+        Parser p = new Parser(command, description);
+        p.executeOnly(this);
+      }
+    } catch (FileNotFoundException e) {
```

Comment:
Perhaps you could create the file if it is not found. I thought that would be better as the user can't do much if you tell them file not found.